src/client/pages/Geo/GeoMap/GeoMap.tsx

```
@@ -1,31 +1,66 @@
+import { defaultBounds } from '@meta/geo'
+
+import { MapContext, useCountryIso } from '@client/hooks'
+import { getCountryBounds } from '@client/pages/Geo/utils/countryBounds'
 
 type Props = {
-  center: google.maps.LatLngLiteral
-  zoom: number
+  center?: google.maps.LatLngLiteral
+  viewport?: google.maps.LatLngBoundsLiteral
+  zoom?: number
 }
 
-const GeoMap: React.FC<Props> = ({ children, center, zoom }) => {
+const GeoMap: React.FC<React.PropsWithChildren<Props>> = (props) => {
+  const { center, viewport, children, zoom } = props
   const ref = useRef<HTMLDivElement>(null)
   const [map, setMap] = useState<google.maps.Map>()
+  const countryIso = useCountryIso()
 
   useEffect(() => {
     if (ref.current && !map) {
-      setMap(
-        new window.google.maps.Map(ref.current, {
-          center,
-          zoom,
-          disableDefaultUI: true,
-          zoomControl: true,
-          rotateControl: true,
-          fullscreenControl: true,
+      const mapSetUp = new window.google.maps.Map(ref.current, {
+        center,
+        zoom: zoom || 4,
+        disableDefaultUI: true,
+        zoomControl: true,
+        rotateControl: true,
+        fullscreenControl: true,
+        mapTypeId: google.maps.MapTypeId.HYBRID,
+        mapTypeControl: true,
+        mapTypeControlOptions: {
+          style: google.maps.MapTypeControlStyle.HORIZONTAL_BAR,
+          position: google.maps.ControlPosition.RIGHT_TOP,
+          mapTypeIds: [google.maps.MapTypeId.ROADMAP, google.maps.MapTypeId.SATELLITE, google.maps.MapTypeId.HYBRID],
+        },
+      })
+      if (!zoom) mapSetUp.fitBounds(viewport)
+      setMap(mapSetUp)
+
+      // Center and focus on the current country
+      getCountryBounds(countryIso)
+        .then((response) => {
+          if (response?.data) {
+            mapSetUp.panTo(response.data.centroid)
+            mapSetUp.fitBounds(response.data.bounds)
+          }
         })
-      )
+        .catch()
```

Possible to avoid try catch and use await and try/catch?

---

at least remove the empty catch ;)

---

There is a try-catch in the fetch function, forgot to clean up the catch here 😅

---

There is a try-catch in the fetch function, forgot to clean up the catch here 😅. It is fixed now.

---

this is resolved for me. @sorja resolve the conversation if you are satisfied with the changes please.